### PR TITLE
feat: no cluster admin for tfy agent proxy

### PR DESCRIPTION
**Pull request by debajyoti-truefoundry**

### CODE REVIEW

File: charts/tfy-agent/values.yaml [hunk 1]
```diff
@@ -432,6 +432,66 @@ tfyAgentProxy:
     ## 
     name: ""
 
+  clusterRole:
+    ## @param tfyAgentProxy.clusterRole.enable Create cluster role.
+    enable: true
+    ## @param tfyAgentProxy.clusterRole.rules [array] Cluster role Authz rules.
+    rules:
+      - apiGroups: [""]
```
Comment (shubhamrai1993) [inline charts/tfy-agent/values.yaml:440]: Should these roles be in values or part of the manifest?

File: charts/tfy-agent/values.yaml [hunk 2]
```diff
@@ -432,6 +432,66 @@ tfyAgentProxy:
...
+        resources: ["statefulsets/scale"]
+        verbs: ["patch"]
+
+      - apiGroups: ["networking.istio.io"]
+        resources: ["virtualservices"]
+        verbs: ["list"]
+
+      - apiGroups: ["keda.sh"]
+        resources: ["scaledobjects"]
+        verbs: ["get", "update"]
```
Comment (chiragjn) [inline charts/tfy-agent/values.yaml:474]: Should we also have `list` wherever we have `get`?

Comment (debajyoti-truefoundry) [reply charts/tfy-agent/values.yaml:478]: list allows you to discover new resources. get does not, you need to know the name.

File: charts/tfy-agent/values.yaml [hunk 3]
```diff
@@ -432,6 +432,70 @@ tfyAgentProxy:
...
+      - apiGroups: [""]
+        resources: ["secrets"]
+        verbs: ["create", "patch"]
+
+      - apiGroups: [""]
+        resources: ["pods"]
+        verbs: ["delete"]
+
+      - apiGroups: [""]
+        resources: ["pods/log", "events", "nodes", "services", "persistentvolumes"]
```
Comment (chiragjn) [inline charts/tfy-agent/values.yaml:457]: do we need v1/apps Deployment somewhere?

File: charts/tfy-agent/values.yaml (hunk 1 above)
Comment (debajyoti-truefoundry) [reply charts/tfy-agent/values.yaml:440]: I am keeping it in values to be easily discoverable and fixable.

File: charts/tfy-agent/values.yaml (hunk 3 above)
Comment (debajyoti-truefoundry) [reply charts/tfy-agent/values.yaml:457]: no

File: charts/tfy-agent/values.yaml [hunk 4]
```diff
@@ -432,6 +432,70 @@ tfyAgentProxy:
...
+        resources: ["rollouts", "workflowtemplates"]
+        verbs: ["get"]
+
+      - apiGroups: ["argoproj.io"]
+        resources: ["workflows"]
+        verbs: ["create", "get", "patch"]
+
+      - apiGroups: ["kubeflow.org"]
+        resources: ["notebooks"]
+        verbs: ["get", "update"]
```
Comment (chiragjn) [inline charts/tfy-agent/values.yaml:498]: Does this need create or update?

https://github.com/truefoundry/servicefoundry-server/blob/3a68b0145add8651f0d4f54559c6fa493dee528c/src/application/controllers/NotebookController.ts#L73

https://github.com/truefoundry/servicefoundry-server/blob/3a68b0145add8651f0d4f54559c6fa493dee528c/src/cluster/K8sClient.ts#L183

- Does update cover the pause/resume case? or do we do some sort of patch op?

Comment (debajyoti-truefoundry) [reply charts/tfy-agent/values.yaml:498]: update covers everything.

**Review by chiragjn — APPROVED** [review]

**Review by akashg3627 — APPROVED** [review]

**Pull request merged**